### ISSUE TITLE
NodeEditorAction eats hotControl

### ISSUE THREAD

**Issue by dashrava** [Siccity/xNode#45]
**Issue:**
It seems that NodeEditorAction (NEA) is eating the hotControl before the button click is registered.

I've narrowed it down to the below section of NEA, specifically setting the hotControls to zero, but I don't know how to properly fix it myself without potentially breaking something else:

```
else if (!IsHoveringNode) {
	// If click outside node, release field focus
	if (!isPanning) {

		// I've got no idea which of these do what, so we'll just reset all of it.
		GUIUtility.hotControl = 0;
		GUIUtility.keyboardControl = 0;
		EditorGUIUtility.editingTextField = false;
		EditorGUIUtility.keyboardControl = 0;
		EditorGUIUtility.hotControl = 0;
	}
	if (NodeEditorPreferences.GetSettings().autoSave) AssetDatabase.SaveAssets();
}
```

**Situation:**
I'm trying to add a global controls box in the corner of the xNode window using my graph's custom editor's OnGUI.

![unknown](https://user-images.githubusercontent.com/17537754/41591583-d2569d1e-73c2-11e8-882c-8650090a8de2.png)

Example implementation: https://github.com/dashrava/xNode/tree/hotControlIssue
(Usage: Open graph.asset, try to press button before and after panning.)

Relevant code bit:
```
[CustomNodeGraphEditor(typeof(NewNodeGraph))]
public class NewNodeGraphEditor : NodeGraphEditor
{

    public override void OnGUI()
    {
        base.OnGUI();

        if (GUILayout.Button("this button doesn't log anything unless you pan or modify NEA"))
        {
            Debug.Log("CLICK");
        }
    }
}
```

**Further information:**
I tried to call use() on the current event when the graph editor's OnGUI button is clicked, but it seems that NEA has priority and clears the control before the button click goes through properly.

The control box buttons (used to zoom in the below gif) work after panning:
![nea](https://user-images.githubusercontent.com/17537754/41591598-df177082-73c2-11e8-8d3f-b7e1692d8c60.gif)

**Considerations:**
Is it necessary to reset the hotControl in the first place?

The control box obviously isn't `IsHoveringNode`.
Is there a way to check that the click is specifically on the editor's background?

**Comment by dashrava** [comment]
Added link to example branch for easier repro. Tested with fresh fork of the master branch (on Unity 2018.1)

**Comment by Siccity** [comment]
Fixed ba3c8bd
